File: frontend/youtube-clone-ui/src/app/dto/video-dto.ts
export interface VideoDto {
  id: string;
  videoId: string;
  userId: string;
  title: string;
  description: string;
  likes: number;
  dislikes: number;
  viewCount: number;
  tags: string[];
  videoStatus: string;
  videoUrl: string;
  thumbnailUrl: string;
  publishedDateAndTime: string;
  username: string;
  userDisplayName: string;
  userPicture: string;
}

export interface WatchedVideoDTO extends VideoDto {
  watchedOn: string | number | Date;
}

export interface UserDto {
  id: string;
  firstName: string;
  lastName: string;
  name: string;
  picture: string;
  sub: string;
  email: string;
  nickname: string;
  subscribedToUsers: any[];
  subscribers: any[];
  videoHistory: any[];
  likedVideos: any[];
  dislikedVideos: any[];
}

export interface CommentDTO {
  id: string;
  text: string;
  userId: string;
  likes: number;
  dislikes: number;
  videoId: string;
  picture: string;
  username: string;
  commentCreatedTime: number;
}

export interface VideoStatus {
  name: string;
  code: string;
}

File: backend/src/main/java/com/karthick/youtubeclone/dto/VideoDTO.java
package com.karthick.youtubeclone.dto;

import com.karthick.youtubeclone.entity.VideoStatus;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.util.Set;

@Data
@NoArgsConstructor
@AllArgsConstructor
public class VideoDTO {

    private String id;
    private String title;
    private String description;
    private Set<String> tags;
    private VideoStatus videoStatus;
    private String videoUrl;
    private String thumbnailUrl;
    private Long likes;
    private Long dislikes;
    private Long viewCount;
    private UserDTO userDTO;


}
